crates/router/src/core/errors.rs

```diff
@@ -309,6 +309,8 @@ pub enum VaultError {
     MissingRequiredField { field_name: &'static str },
     #[error("The card vault returned an unexpected response: {0:?}")]
     UnexpectedResponseError(bytes::Bytes),
+    #[error("Failed to update in PMD table")]
+    UpdateInPMDTableFailed,
```

Comment:
Either follow the Rust naming convention:

```suggestion
    UpdateInPmdTableFailed,
```

or use the complete name:

```suggestion
    UpdateInPaymentMethodDataTableFailed,
```

And please be verbose in the error messages, do not use abbreviations there.